src/sync/impl/sync_metadata.cpp

```diff
@@ -334,22 +338,31 @@ SyncUserMetadata::SyncUserMetadata(Schema schema, SharedRealm realm, RowExpr row
 , m_row(row)
 { }
 
+void SyncUserMetadata::checkValidOrThrow(const std::string operation) const
```

Comment:
This should just take a StringData or const char* to avoid constructing a std::string every time it's called.

Reply:
The usual name for this function would be just `verify_attached()`.